### ISSUE TITLE
What would need to be done to utilize gpu?

### ISSUE THREAD

**Issue by CptanPanic** [xserrat/docker-facebook-demucs#21]
I was wondering what would need to be done to make this version able to utilize the nvidia gpu? I am guessing would need to start Dockerfile using a nvidia image, and then install python, and everything else? Then would just need to add option in Dockerfile to download the gpu model?

**Comment by xserrat** [comment]
Hi @CptanPanic ,

I guess we could add a Dockerfile that starts from the Nvidia image. Do you have the Nvidia image you're referencing?

Thanks for your contribution :)

**Comment by ForceConstant** [comment]
Maybe something like nvidia's pytorch image? nvcr.io/nvidia/pytorch:latest as I think demucs uses pytorch.

**Comment by xserrat** [comment]
I'll take a look because it would be great to run the demucs in different modes 💪🏼

**Comment by aavetis** [comment]
any branch that might have been started that i can contribute to?

**Comment by xserrat** [comment]
@aavetis Thanks for contributing to the project! 🚀 I started the development in my local environment but didn't get an early version that worked, I got this error building the image `nvcr.io/nvidia/pytorch:22.08-py3`:
```
stderr: nvidia-container-cli: initialization error: load library failed: libnvidia-ml.so.1: cannot open shared object file: no such file or directory: unknown
```
It can be related to this issue https://github.com/NVIDIA/libnvidia-container/issues/135 but I've not digged into the issue itself yet.

So great to see a pull request with a first approach on that :) I'll take a look and we can discuss there the changes! Thanks again and welcome to the `docker-facebook-demucs` repo!
